smart_open/s3.py

```diff
@@ -1137,16 +1137,11 @@ def __init__(
 
         if writebuffer is None:
             self._buf = io.BytesIO()
+        elif not writebuffer.seekable():
+            raise ValueError('writebuffer needs to be seekable')
         else:
             self._buf = writebuffer
 
-        self._total_bytes = 0
-
-        #
-        # This member is part of the io.BufferedIOBase interface.
-        #
-        self.raw = None
```

Comment:
`_BufferedIOMixin` (not used here) would expect `self._raw` which would be accessed via the [`raw`](https://github.com/python/cpython/blob/v3.13.2/Lib/_pyio.py#L834) property